sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/hints.scala

```diff
@@ -25,7 +25,7 @@ import org.apache.spark.sql.internal.SQLConf
  * should be removed This node will be eliminated post analysis.
  * A pair of (name, parameters).
  */
-case class UnresolvedHint(name: String, parameters: Seq[String], child: LogicalPlan)
+case class UnresolvedHint(name: String, parameters: Seq[Any], child: LogicalPlan)
```

Comment:
To support multiple parameters in `hint`, does it make sense to do it like `df.hint("hint", "1, 2, c")`? We can use our Parser to parse this parameter string.